### PR TITLE
dts: nordic: adding FICR binding file and macro define

### PR DESCRIPTION
**Pull request by ioannisg**
Adding a binding .yaml file for Nordic FICR and the
corresponding macro definition for NRF_FICR in
nrfx_config_nrf9160.h header file.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: dts/bindings/arm/nordic,nrf-ficr.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
+title: Nordic FICR (Factory Information Configuration)
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply dts/bindings/arm/nordic,nrf-ficr.yaml]: Yes, we can do that - thanks for the feedback

File: ext/hal/nordic/nrfx_config_nrf9160.h [hunk 2]
```diff
@@ -46,6 +46,11 @@
         ((NRF_CLOCK_Type *)DT_NORDIC_NRF_CLOCK_CLOCK_BASE_ADDRESS)
 #endif
 
+#ifdef DT_NORDIC_NRF_FICR_FICR_BASE_ADDRESS
+#define NRF_FICR \
+        ((FICR_INFO_Type*) DT_NORDIC_NRF_FICR_FICR_BASE_ADDRESS)
```
[the remark this answers is not in the document]
Comment (ioannisg) [reply ext/hal/nordic/nrfx_config_nrf9160.h]: Fixed, thanx

[the remark this answers is not in the document]
Comment (anangl) [reply ext/hal/nordic/nrfx_config_nrf9160.h]: Still missing space before `*`.

[the remark this answers is not in the document]
Comment (ioannisg) [reply ext/hal/nordic/nrfx_config_nrf9160.h]: Ah, so sorry - I fixed only the extra space before `D`

**Review by anangl — APPROVED** [review]

**Pull request merged**